src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -260,10 +268,16 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
       // Create the bucket in leo's google project and populate with initialization files.
       // ACLs are granted so the cluster service account can access the bucket at initialization time.
       initBucketPath <- initializeBucket(userEmail, googleProject, clusterName, initBucketName, clusterRequest, serviceAccountInfo, serviceAccountKeyOpt)
+      //Get list of groups to give staging bucket access to
+      groupstagingBucketAcl <- serviceAccountProvider.listGroupsStagingBucketReaders(userEmail)
+      userstagingBucketAcl <- serviceAccountProvider.listUsersStagingBucketReaders(userEmail)
```

Comment:
Maybe we should think about defaults. If `listGroupsStagingBucketReaders` and `listUsersStagingBucketReaders` both return an empty list, then the cluster will be unusable because the pet can't access the staging bucket. Maybe AT A MINIMUM we should always grant access to the pet, and these methods can be used for adding to that. It'd be nice if implementors could choose to _not_ implement `listGroupsStagingBucketReaders` and `listUsersStagingBucketReaders`, and have a sensible default in Leo.

Reply:
s/pet/cluster-SA and notebook-SA, but i'd be inclined to have that as the default-impl in ServiceAccountProvider rather than specifying them here.

Reply:
Yeah sorry, that's what I was thinking too: add a default implementation of `ServiceAccountProvider.listUsersStagingBucketReaders()` that returns 
```
Future.sequence(List(getClusterServiceAccount(...), getNotebookServiceAccount(...))).map(_.flatten)
```
(I think that works.)

`listGroupsStagingBucketReaders` can probably just default to `Future.successful(List.empty)`.

Reply:
We currently give LeoSA owner access to the bucket and clusterSA reader access. 

https://github.com/DataBiosphere/leonardo/blob/c56b02e21c24205ca344d135de4f1f5910859dd6/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala#L339

I will move this to the default implementation of ServiceAccountProvider and also give notebookSA reader access to the bucket.

Reply:
Faces: the default behavior is specified correctly in the DAO. Moving it to the provider would be nicer, but posed some problems. Going to leave it in the DAO for now (might refactor it later to be in the provider).